source/agora/network/NetworkManager.d

```diff
@@ -235,35 +241,34 @@ public class NetworkManager
                 return size_t.max;
         }
 
-        try
-        {
-            auto node_pair = this.peers.byValue
-                .map!(node => tuple(getHeight(node), node))
-                .filter!(pair => pair[0] != ulong.max)  // request failed
-                .reduce!((a, b) => a[0] > b[0] ? a : b);
+        auto node_pair = this.peers.byValue
+            .map!(node => Pair(getHeight(node), node))
+            .filter!(pair => pair.height != ulong.max)  // request failed
+            .each!(pair => node_pairs ~= pair);
 
-            const highest_block = node_pair[0];
-            if (highest_block < block_height)
-                return;  // we're up to date
+        node_pairs.sort!((a, b) => a.height < b.height);
 
-            logInfo("Retrieving latest blocks..");
-            auto node = node_pair[1];
+        LNextNode: foreach (pair; node_pairs.retro) try
```

Comment:
You could just change the ordering and wouldn't need `retro`, no ?

Reply:
LOL